canvasapis.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import requests
+import urllib
+
+
+class GroupsForSections:
+    """
+    This class makes the api calls to create groups for sections. TODO: this class will be taken out into a separate
+    module in a different repo
+    """
+    HTTP_METHOD_POST = 'Post'
+    HTTP_METHOD_GET = 'Get'
+    AUTHORIZATION = 'Authorization'
+    BEARER = 'Bearer '
+    MIME_TYPE_JSON = 'application/json'
+    CONTENT_TYPE = 'Content-type'
+
+    def __init__(self, canvas_token, canvas_url):
+        self.canvas_token = canvas_token
+        self.canvas_url = canvas_url
+
+    def create_group_category(self, group_category_name, course_id):
+        """
+        Creates a group category with a given name for a particular course, In case of exceptions it throws it to
+        calling method
+
+        :param String group_category_name:
+        :return: HttpResponse response
+        """
+
+        url = self.canvas_url + '/api/v1/courses/' + course_id + '/group_categories?name=' + urllib.quote_plus(
+                group_category_name)
+        try:
+            response = self.make_api_call(url, self.HTTP_METHOD_POST)
+        except:
+            raise
+
+        return response
+
+    def sections_for_course(self, course_id, next_page_url=None):
+        """
+        get sections in a course, throws the exception to the calling method
+        :param String next_page_url:
+        :return: HttpResponse response
+        """
+
+        if next_page_url is not None:
+            url = next_page_url
+        else:
+            url = self.canvas_url + '/api/v1/courses/' + course_id + '/sections?per_page=100'
+
+        try:
+            response = self.make_api_call(url, self.HTTP_METHOD_GET)
+        except:
+            raise
```

Comment:
Yes, it raises an exception ( which is RequestException) that is throw in the `api_handler()' line number 147, 146 . Below is the stack trace of a api call when i have sent 'o' instead of url

```2016-07-01 13:43:51,727 - root - ERROR - getting USERS list has erroneous response Invalid URL 'o': No schema supplied. Perhaps you meant http://o?
Traceback (most recent call last):
  File "groupsforsections.py", line 91, in get_users_in_section
    response = groups_for_section.users_in_section(section_id, next_page_url)
  File "/Users/pushyami/src/itsProjects/canvas_groups_for_sections/canvasapis.py", line 73, in users_in_section
    response = self.make_api_call('o', self.HTTP_METHOD_GET)
  File "/Users/pushyami/src/itsProjects/canvas_groups_for_sections/canvasapis.py", line 122, in make_api_call
    response = self.api_handler(url, http_method)
  File "/Users/pushyami/src/itsProjects/canvas_groups_for_sections/canvasapis.py", line 149, in api_handler
    raise request_exception
MissingSchema: Invalid URL 'o': No schema supplied. Perhaps you meant http://o?```